velox/exec/PrefixSort.cpp

```diff
@@ -120,7 +125,7 @@ PrefixSortLayout PrefixSortLayout::makeSortLayout(
   uint32_t numNormalizedKeys = 0;
   const uint32_t numKeys = types.size();
   std::vector<uint32_t> prefixOffsets;
-  std::vector<PrefixSortEncoder> encoders;
+  std::vector<std::shared_ptr<PrefixSortEncoder>> encoders;
```

Comment:
Could you explain why we are making this change?

Reply:
Because for hugeint, we need reserve precision and scale, so it extends PrefixSortEncoder, to save it, we need to make it shared_ptr